### PR TITLE
feat: Add support for extra fields on Person details

### PR DESCRIPTION
**Pull request by pawelsz-rb**
## Description of the change
(duplicate of https://github.com/rollbar/rollbar-go/pull/91)
This is to help achieve parody with the PHP SDK which allows for
additional fields to be specified on the Person object. We use this to
provide details like what account a specific user belongs to or other
additional internal metadata that is helpful when debugging user/account
level errors.

This is mainly mirroring what I saw in the PHP SDK here: 

https://github.com/rollbar/rollbar-php/blob/89c9069e8532acb638f2fb9efd36b529ec25354f/src/Payload/Person.php#L68-L80

```php
    public function serialize()
    {
        $result = array(
            "id" => $this->id,
            "username" => $this->username,
            "email" => $this->email,
        );
        foreach ($this->extra as $key => $val) {
            $result[$key] = $val;
        }
        
        return $this->utilities()->serializeForRollbarInternal($result, array_keys($this->extra));
    }
```

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

## Related issues

From what I've seen there are not any GitHub issues for this.

### Development

- [x] Lint rules pass locally
      _(I don't see any editor config or linting documentation in the README or anywhere else in the repo?)_
- [x] The code changed/added as part of this pull request has been covered with tests
      _(This code should be covered by existing unit tests, but happy to add a specific test for this code path)_
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers assigned
- [x] Issue from task tracker has a link to this pull request
- [x] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: rollbar.go [hunk 1]
```diff
@@ -231,8 +231,8 @@ func SetCheckIgnore(checkIgnore func(string) bool) {
 // SetPerson information for identifying a user associated with
 // any subsequent errors or messages. Only id is required to be
 // non-empty.
-func SetPerson(id, username, email string) {
-	std.SetPerson(id, username, email)
+func SetPerson(id, username, email string, opts personOption) {
```
Comment (ijsnow) [inline rollbar.go:234]: Should this be variadic as well to match the client's signature?

**Review by ijsnow — COMMENTED** [review]
Looks good, just one question

File: rollbar.go (hunk 1 above)
Comment (pawelsz-rb) [reply rollbar.go]: good catch, I guess we did not have tests for SetPerson on rollbar level

**Review by ijsnow — APPROVED** [review]
Looks good!

**Pull request merged**